packages/react/src/components/Dialog/Dialog.tsx

```diff
@@ -177,6 +177,21 @@ class Dialog extends AutoControlledComponent<ReactProps<DialogProps>, DialogStat
     },
   })
 
+  /**
+   * In case it generates an id, its uses will ensure that the id on the header
+   * and the value of aria-labelledBy on the dialog will be the same.
+   */
+  private getDefaultAriaLabelledBy() {
+    const { header } = this.props
+    if (this.props['aria-label'] || this.props['aria-labelledby'] || !header) {
+      return undefined
+    }
+    if (typeof header === 'string' || !header['id']) {
```

Comment:
if the header is a string type there is never gonna be `header[id]`, isn't it? or I might misunderstand smth

Reply:
does it cover the case when header is object but does not have `id` ?

Reply:
oh probably you are right. `!header['id']` should cover both. great!